src/MouseMoveProvider.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+import React, { Component, PropTypes } from 'react';
+import { uuid } from './utils';
+import invariant from 'invariant';
+
+type Subscription = { uuid: string, callback: () => void };
+type SubscriptionArray = [Subscription];
+
+export default class MouseMoveProvider extends Component {
+  subs: SubscriptionArray = [];
+  hasSubs: Boolean = false;
+  initialOffset: Number = window.pageYOffset;
+
+  static propTypes = {
+    children: PropTypes.node
+  };
+
+  static childContextTypes = {
+    mouseMove: PropTypes.object.isRequired
+  };
+
+  getChildContext() {
+    const { subscribe } = this;
+    return {
+      mouseMove: {
+        subscribe
+      }
+    };
+  }
+
+  componentWillMount() {
+    const { children } = this.props;
+    invariant(children === null || React.Children.count(children) === 1, 'A <MouseMoveProvider> may have only one child element');
+  }
+
+  componentWillUnmount() {
+    this.subs = [];
```

Comment:
We shouldn't need to reassign this on unmount, it will get garbage collected along with the component instance.